src/server/verification/processor/EnrollmentProcessor.js

```diff
@@ -0,0 +1,117 @@
+// @flow
+import { noop } from 'lodash'
+
+import Config from '../../server.config'
+import { GunDBPublic } from '../../gun/gun-middleware'
+import AdminWallet from '../../blockchain/AdminWallet'
+
+import { recoverPublickey } from '../../utils/eth'
+
+import { type IEnrollmentProvider } from './typings'
+
+import EnrollmentSession from './EnrollmentSession'
+import ZoomProvider from './provider/ZoomProvider'
+
+class EnrollmentProcessor {
+  gun = null
+  storage = null
+  adminApi = null
+  keepEnrollments = null
+
+  _provider = null
+
+  get provider() {
+    const { _provider } = this
+
+    if (!_provider) {
+      throw new Error(`Provider haven't registered.`)
+    }
+
+    return _provider
+  }
+
+  constructor(storage, Config, adminApi, gun) {
+    const { keepFaceVerificationRecords } = Config
+
+    this.gun = gun
+    this.storage = storage
+    this.adminApi = adminApi
+    this.keepEnrollments = keepFaceVerificationRecords
+  }
+
+  registerProvier(provider: IEnrollmentProvider): void {
+    this._provider = provider
+  }
+
+  validate(user: any, enrollmentIdenfitier: string, payload: any) {
+    const { sessionId } = payload || {}
+    const { provider } = this
+
+    if (!user || !enrollmentIdenfitier || !payload || !sessionId || !provider.isPayloadValid(payload)) {
+      throw new Error('Invalid input')
+    }
+  }
+
+  async enroll(user: any, enrollmentIdenfitier: string, payload: any): Promise<any> {
+    const { provider, storage, adminApi, gun } = this
+    const session = new EnrollmentSession(user, provider, storage, adminApi, gun)
+
+    return session.enroll(enrollmentIdenfitier, payload)
+  }
+
+  async enqueueDisposal(enrollmentIdentifier, signature) {
+    const { provider } = this
+    const recovered = recoverPublickey(signature, enrollmentIdentifier, '')
+
+    if (recovered.substr(2) !== enrollmentIdentifier.toLowerCase()) {
+      throw new Error(
+        `Unable to enqueue enrollment '${enrollmentIdentifier}' disposal: ` +
+          `SigUtil unable to recover the message signer`
+      )
+    }
+
+    const enrollmentExists = await provider.enrollmentExists(enrollmentIdentifier)
+
+    if (enrollmentExists) {
+      // TODO: enqueue enrollmentIdentifier to the corresponding mongo collection using storage
+      // add current timestamp to each collection item
+    }
+  }
+
+  async disposeEnqueuedEnrollments(onProcessed: (identifier: string, exception?: Error) => void): Promise<void> {
+    noop(onProcessed) // eslint / lgtm stub
+    // TODO
+
+    // 1. get all items from enqueued enrollment identifiers collection
```

Comment:
we could be running multiple nodes of GoodServer, so this needs to happen atomically (so two servers dont try to delete same record)
it can be done either by using mongo findOneAndUpdate or use the queueMongo lock that we use to synchronize blockchain calls across servers